cs/src/core/Allocator/AllocatorBase.cs

```diff
@@ -1366,6 +1368,7 @@ public void AsyncFlushPages(long fromAddress, long untilAddress)
         public void AsyncFlushPagesToDevice(long startPage, long endPage, long endLogicalAddress, IDevice device, IDevice objectLogDevice, out CountdownEvent completed)
         {
             int totalNumPages = (int)(endPage - startPage);
+            var localSegmentOffsets = new long[SegmentBufferSize];
```

Comment:
Wondering if we could prevent allocation for non-GenericAllocator case.

Reply:
Hmm, good point, yes, that's doable with some refactoring